### PR TITLE
CLOUDP-70039: mongocli atlas maintenanceWindow(s) update

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-70039

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas maintenanceWindows update --hourOfDay 1 --dayOfWeek 1 
Maintenance window updated.
```

Screenshot 2020-09-08 at 14 55 14

### CODE REVIEW

File: internal/cli/atlas/maintenance/update.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+func (opts *UpdateOpts) Run() error {
+	err := opts.store.UpdateMaintenanceWindow(opts.ConfigProjectID(), opts.newMaintenanceWindow())
+	if err != nil {
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
```
Comment (gssbzn) [inline internal/cli/atlas/maintenance/update.go:54]: duplicate block?

File: internal/store/maintenance.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+//go:generate mockgen -destination=../mocks/mock_maintenance.go -package=mocks github.com/mongodb/mongocli/internal/store MaintenanceWindowUpdater
+
+type MaintenanceWindowUpdater interface {
+	UpdateMaintenanceWindow(string, *atlas.MaintenanceWindow) error
+}
+
+// UpdateMaintenanceWindow encapsulates the logic to manage different cloud providers
+func (s *Store) UpdateMaintenanceWindow(projectID string, maintenanceWindow *atlas.MaintenanceWindow) error {
```
Comment (gssbzn) [inline internal/store/maintenance.go:32]: [non-blocking] FYI ops manager works way different from Atlas so this file and methods may need to reflect that when we get there

File: internal/cli/atlas/maintenance/update.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	fmt.Print(updateTemplate)
```
Comment (gssbzn) [inline internal/cli/atlas/maintenance/update.go:56]: [nit] what happened to using the Output?

**Review by gssbzn — CHANGES_REQUESTED** [review]
two small nits and one non-blocking comment

File: internal/cli/atlas/maintenance/update.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+var updateTemplate = "Maintenance window updated.\n"
+
+func (opts *UpdateOpts) Run() error {
+	err := opts.store.UpdateMaintenanceWindow(opts.ConfigProjectID(), opts.newMaintenanceWindow())
+	if err != nil {
+		return err
+	}
+	opts.Print(updateTemplate)
+	return nil
```
Comment (gssbzn) [inline internal/cli/atlas/maintenance/update.go:50]: [nit] ignored error, also you can pass nil if update doesn't return anything
```suggestion
	return opts.Print(nil)
```

File: internal/cli/atlas/maintenance/update.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+var updateTemplate = "Maintenance window updated.\n"
+
+func (opts *UpdateOpts) Run() error {
+	err := opts.store.UpdateMaintenanceWindow(opts.ConfigProjectID(), opts.newMaintenanceWindow())
+	if err != nil {
+		return err
+	}
+	opts.Print(updateTemplate)
+	return opts.Print(nil)
```
Comment (gssbzn) [inline internal/cli/atlas/maintenance/update.go:50]: duplicate?

**Review by gssbzn — CHANGES_REQUESTED** [review]

File: internal/cli/atlas/maintenance/update.go (hunk 5 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/maintenance/update.go]: 😿

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**